tests/src/python/test_qgscolorramp.py

```diff
@@ -288,11 +288,14 @@ def testQgsRandomColorRamp(self):
         self.assertEqual(cloned.type(), 'randomcolors')
 
         # test with pregenerated colors
-        for i in range(10000):
```

Comment:
What's the rationale behind lowering this size?

Reply:
Since I changed the inner loop to do `n` iterations rather than just 10, I did not want to increase the amount of work. The previous test did 100000 calls to color(); the new one does around 5000.

I didn't quite understand why 10000 repetitions were needed - maybe there's something in the random color generation that could generate invalid (as in `not c.isValid()`) previously?

Reply:
> maybe there's something in the random color generation that could generate invalid (as in not c.isValid()) previously?

exactly -- we had a few bugs in the past where the range of r/g/b values generated fell outside of acceptable ranges sometimes, which lead to invalid colors. What you've got here is probably fine though!